src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -162,11 +306,15 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
                     string? repoCandidate = doc.DocumentNode.SelectSingleNode(path)?.GetAttributeValue("href", string.Empty);
                     if (!string.IsNullOrEmpty(repoCandidate))
                     {
-                        PackageURL repoPurl = GitHubProjectManager.ExtractGitHubPackageURLs(repoCandidate).ToList().FirstOrDefault();
-                        mapping.Add(repoPurl, 1.0F);
-                        return mapping;
+                        var candidate = ExtractGitHubPackageURLs(repoCandidate).FirstOrDefault();
+                        if (candidate != null)
+                        {
+                            mapping.Add((PackageURL)candidate, 1.0F);
+                            
```

Comment:
Whitespace

Reply:
@gfs Would `mapping.Add(candidate as PackageURL, 1.0F);` work too? Is there a different because a c-style cast and the `as` operator?

Reply:
I think (but haven't tested) that that would work as well if you like how it looks.

Reply:
The difference appears to be:

* as operator: failure to cast at runtime returns null. Compile time checks for types. 
* Casting: failure to cast at runtime throws an exception. Compile time checks for types. 

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast